borg/_hashindex.c

```diff
@@ -77,6 +79,7 @@ static int hash_sizes[] = {
 
 #define HASH_MIN_LOAD .25
 #define HASH_MAX_LOAD .75  /* don't go higher than 0.75, otherwise performance severely suffers! */
+#define HASH_MAX_EFF_LOAD .85
```

Comment:
I'll check if I find again where I got the 0.85 from.
But just from a rough estimation, if we just had some few empty buckets distributed more or less evenly over the hashtable that already help tremendously compare to full table scans with no empty buckets.
0 empty -> full scan
10 empty evenly distr. -> 10% scan
100 -""- -> 1% scan
...

So, gut feeling is that anything <= 99% is already ok.
We aren't searching for the super high performance, good performance is already ok.